### PR TITLE
V2/remove-v1

### LINKED ISSUE #486: [TODO]: move `chemprop/v2` to `chemprop`
Let's not forget to move all legacy code out of the source distribution when we release it. While we're still in the process of reimplementing features from `v1`, I propose we at least keep the original source inside the repo under `$CHEMPROP_REPO/legacy/chemprop`. I.e., outside of the actual `chemprop` distribution.

### LINKED ISSUE #489: Update package authors after v2 release
Update package authors lists to include all major contributors in all the relevant locations:
* `pyproject.toml`
* `docs/*`
* `LICENSE.txt`
* ~`setup.cfg`~ (will be deleted, I think)
* ~`setup.py`~ (has been deleted)

### PR DESCRIPTION
**Pull request by kevingreenman**
## Description

- As discussed in our meeting, since we are now in beta, we are removing all remaining v1 code from this branch and moving the v2 code up to the package level.
- I've also:
  - Changed the version / release to `2.0.0b1` and updated the `.bumpversion.cfg` file to reflect that we no longer have a `setup.py` or `setup.cfg` file.
  - Updated package authors, etc - similar to #532 and #533 for v1
  - Changed location of v2 tests directory for CI
  - Updated all import statements (including in examples, tests, etc.) to reflect new location of v2 files 
  - Added a top-level `__init.py__` file as suggested in #517
  - Fixed `Metric` not defined error in integration test

## Questions
* What else should be updated in this PR to ensure all of our v2 code works as well as it did before this PR and to enable a clean installation for our beta testers?

## Relevant issues
* Fixes #486
* Fixes #489

### CODE REVIEW

**Comment by kevingreenman** [comment]
@davidegraff I'm a little confused about what is going on with this `Metric` not defined error in one of the integration tests, and why it's happening now but wasn't before moving the v2 code up a level. Any help debugging would be appreciated.

Other than that, I think this PR is ready to look at. Please let me know if you think there's any other changes that should be part of this PR.

File: chemprop/__init__.py [hunk 1]
```diff
@@ -1,22 +1,13 @@
...
-# import chemprop.models
-# import chemprop.train
-# import chemprop.uncertainty
-# # import chemprop.web
+import chemprop.cli
+import chemprop.data
+import chemprop.featurizers
+import chemprop.models
+import chemprop.nn
+import chemprop.utils
```
Comment (davidegraff) [inline chemprop/__init__.py:6]: same goes for the rest of the file:
```suggestion
from . import data, featurizers, models, nn, utils
```

File: chemprop/cli/utils/utils.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 from typing import Any, Type
 
-from chemprop.v2.nn import loss, predictors
+from nn import loss, predictors
```
Comment (davidegraff) [inline chemprop/cli/utils/utils.py:3]:
```suggestion
from chemprop.nn import loss, predictors
```

**Review by davidegraff — COMMENTED** [review]
only small style comments--will address bugs in different comments

**Comment by davidegraff** [comment]
> @davidegraff I'm a little confused about what is going on with this `Metric` not defined error in one of the integration tests, and why it's happening now but wasn't before moving the v2 code up a level. Any help debugging would be appreciated.
> 
> Other than that, I think this PR is ready to look at. Please let me know if you think there's any other changes that should be part of this PR.

So the stack trace seems to be:
```
tests/integration/test_regression_mol.py:14: in <module>
    from chemprop import featurizers, models, nn
chemprop/__init__.py:4: in <module>
    import chemprop.models
chemprop/models/__init__.py:1: in <module>
    from .model import MPNN
chemprop/models/model.py:10: in <module>
    from chemprop.metrics import Metric
chemprop/metrics.py:9: in <module>
    from chemprop.nn.loss import *
chemprop/nn/__init__.py:4: in <module>
    from .predictors import *
chemprop/nn/predictors.py:[58]: in <module>
    class _FFNPredictorBase(Predictor, HyperparametersMixin):
chemprop/nn/predictors.py:63: in _FFNPredictorBase
    _default_metric: Metric
E   NameError: name 'Metric' is not defined
```
and my interpretation is that the glob statement `import *` is producing some cyclic dependency that we didn't previously have...

**Comment by kevingreenman** [comment]
@davidegraff yeah, it looks like `chemprop.nn.predictors` depends on `chemprop.metrics` and `chemprop.metrics` depends on `chemprop.nn.loss`, which is a part of `chemprop.nn` that includes `chemprop.nn.predictors`

The error was there before the top-level `__init__.py` was present, so that's not the problem.

**Comment by davidegraff** [comment]
how about moving `metrics` down into `chemprop.nn`?

**Comment by kevingreenman** [comment]
I still get circular imports after moving it into `chemprop.nn`

**Comment by davidegraff** [comment]
I'm getting a different failure now:
```
$ pytest tests/integration/test_regression_mol.py
ERROR tests/integration/test_regression_mol.py::test_quick[mp0-data0] - FileNotFoundError: [Errno 2] No such file or directory: 'tests/data/regression_mol.csv'
ERROR tests/integration/test_regression_mol.py::test_quick[mp1-data0] - FileNotFoundError: [Errno 2] No such file or directory: 'tests/data/regression_mol.csv'
ERROR tests/integration/test_regression_mol.py::test_overfit[mp0-data0] - FileNotFoundError: [Errno 2] No such file or directory: 'tests/data/regression_mol.csv'
ERROR tests/integration/test_regression_mol.py::test_overfit[mp1-data0] - FileNotFoundError: [Errno 2] No such file or directory: 'tests/data/regression_mol.csv'
```

**Comment by kevingreenman** [comment]
oops sorry, I had forgotten to make the change to `__init__.py`. I'm not sure how those test files got deleted, but I just put them back, and the tests pass now. I also noticed one of the example notebooks and the `example.py` needed to have their import of `metrics` updated.

**Review by davidegraff — APPROVED** [review]
Something that just came to mind is that we should clean up the test data folder, but probably better to tackle that in a separate PR

**Pull request merged**